src/compiler/crystal/syntax/parser.cr

```diff
@@ -2654,7 +2653,7 @@ module Crystal
         next_token_skip_space
       end
 
-      unless @token.keyword? && @token.value.in?(:when, :else, :end)
+      unless @token.keyword? && @token.value.in?(Keyword::WHEN, Keyword::ELSE, Keyword::END)
```

Comment:
I believe the first part of the condition is unnecessary. If the value is a `Keyword` member, `Token#keyword?` should always be true, right?

```suggestion
      unless @token.value.in?(Keyword::WHEN, Keyword::ELSE, Keyword::END)
```